### ISSUE TITLE
[question] or [feature request] Importing point clouds from laser scanner for meshing

### ISSUE THREAD

**Issue by TheBricktop** [alicevision/meshroom#872]
Is there a way to import external point cloud for meshing the object ? There are some other programs that kinda work like this for example meshlab but its mindmeltingly slow comparing to other programs (cloud compare is faster in poisson reconstruct) also im currently trying to bake vertex colour from point cloud to mesh and it is stuck at it for hour and still nothing. Basically you cant make texture bigger than 1k or its going to die on you. Even blender had no problem baking from 4m or so mesh to low poly.
So im looking everywhere to see if there is another pipeline for vertex colour transmission on mesh, that doesnt cost the equal value of diamond mine.

**Comment by natowi** [comment]
Sorry, this is not yet supported.
There is this WIP PR https://github.com/alicevision/AliceVision/pull/425 based on PCL, but it will take an indeterminate amount of time to finish.
https://github.com/alicevision/meshroom/issues/764
https://github.com/alicevision/meshroom/issues/864

**Comment by TheBricktop** [comment]
Well ive managed to boost the meshlab speed by assigning more ram to it. But for the 3d registration in alicevision : did You consider using Open3D library by intel? Seem very promising and of course has python interface so it could be not so hard to implement.

**Comment by natowi** [comment]
> Seem very promising and of course has python interface so it could be not so hard to implement.

I agree, Open3D could bring some interesting features to Meshroom. For Point Cloud Registration, the GUI implementation for the initial alignment will require some work. The processing part can relatively easy be wrapped in Meshroom nodes.
I found this [Houdini2open3D plugin](https://github.com/jtomori/houdini2open3d/blob/master/scripts/python/hou2open3d.py) that could be used as additional reference to the open3D documentation and examples.

The dev. resources are limited, so I can not say when the 3d-to-3d PR will be completed.

If Point Cloud Registration (based on Open3D) is something you would be interested in contributing, it would be most welcome.
Here is a rough node implementation reference https://github.com/natowi/meshroom_external_plugins/wiki/2-DevDoc

**Comment by myokeeh** [comment]
I came here specifically to find out if this feature already exists. This would be a nice addition.

**Comment by danmandle** [comment]
Is this now a feature with this commit? https://github.com/alicevision/AliceVision/pull/425

**Comment by crazyrobinhood** [comment]
Any news on this? This post is quite old..

**Comment by Chaosminecraft** [comment]
> Any news on this? This post is quite old..

Idk, I have Polycam on my iPhone and I was thinking that maybe the LiDAR data of there could be quite useful for that stuff, I had an idea that I might cook up something that is compatible with Mehsroom and MeshroomCL (But first I would need to learn how the data needs to be interpreted and how to code for that stuff)

**Comment by natowi** [comment]
So far, only alignment https://github.com/alicevision/AliceVision/pull/1803 has been implemented yet.
https://github.com/alicevision/MeshroomResearch may be interesting if you are interested in implementing something of your own.
